### PR TITLE
Custom Field: Java examples

### PR DESCRIPTION
**Pull request by anezthes**
The Java highlight might be on the wrong part. I think it might be better to highlight the actual custom fields instead of where/how they are used.

cc @rolfsmeds @jouni

### CODE REVIEW

**Comment by rolfsmeds** [comment]
> The Java highlight might be on the wrong part. I think it might be better to highlight the actual custom fields instead of where/how they are used.
> 
> cc @rolfsmeds @jouni

Yeah, I would make DateRangePicker.java the main sample file (uh, can we affect the order, or would that require renaming the file to be alphabetically first?), and highlight the class signature and constructor.

**Comment by jouni** [comment]
> can we affect the order

Asciidoc source order defines it.

**Review by magi42 — COMMENTED** [review]
AsciiDoc OK.

File: src/main/java/com/vaadin/demo/component/customfield/CustomFieldBasic.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  public CustomFieldBasic() {
+    // tag::snippet[]
+    DateRangePicker dateRangePicker = new DateRangePicker();
+    dateRangePicker.setLabel("Enrollment period");
+    dateRangePicker.setHelperText("Cannot be longer than 30 days");
+    add(dateRangePicker);
+
+    Binder<EnrollmentPeriod> binder = new Binder();
+    EnrollmentPeriod enrollmentPeriod = new EnrollmentPeriod();
```
[the remark this answers is not in the document]
Comment (anezthes) [reply src/main/java/com/vaadin/demo/component/customfield/CustomFieldBasic.java]: Fixed.

File: src/main/java/com/vaadin/demo/component/customfield/DateRangePicker.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  public DateRangePicker() {
+    start = new DatePicker();
+    start.setPlaceholder("Start date");
+
+    end = new DatePicker();
+    end.setPlaceholder("End date");
+
+    // aria-label for screen readers
```
[the remark this answers is not in the document]
Comment (anezthes) [reply src/main/java/com/vaadin/demo/component/customfield/DateRangePicker.java:25]: Fixed.

File: frontend/demo/component/custom-field/custom-field-basic.ts [hunk 3]
```diff
@@ -19,8 +18,18 @@ export class Example extends LitElement {
   private binder = new Binder(this, AppointmentModel);
 
   firstUpdated() {
+    // aria-label for screen readers
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/custom-field/custom-field-basic.ts:21]: Fixed.

File: src/main/java/com/vaadin/demo/component/customfield/MoneyField.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  public MoneyField() {
+    amount = new TextField();
+
+    currency = new Select();
+    currency.setItems("AUD", "CAD", "CHF", "EUR", "GBP", "JPY", "USD");
+    currency.setWidth("6em");
+
+    // aria-label for screen readers
```
[the remark this answers is not in the document]
Comment (anezthes) [reply src/main/java/com/vaadin/demo/component/customfield/MoneyField.java:26]: Fixed.

**Comment by anezthes** [comment]
> Asciidoc source order defines it.

pLeAsE hAlP!1!1

**Comment by rolfsmeds** [comment]
> > Asciidoc source order defines it.
> 
> pLeAsE hAlP!1!1

Just make sure the file you want to emphasize (show by default) comes first in the source (within its group).

**Comment by jouni** [comment]
```
[.example]
--
[source] ← this will be shown first
----
...
----

[source]
----
...
----
--
```

**Review by jouni — APPROVED** [review]

**Pull request merged**